document/src/main/java/com/ritense/document/service/impl/JsonSchemaDocumentSearchService.java

```diff
@@ -158,44 +244,238 @@ private void addJsonFieldPredicates(CriteriaBuilder cb, Root<JsonSchemaDocument>
+            .collect(toList())
+            .toArray(Predicate[]::new);
+
+        if (searchRequest.getSearchOperator() == SearchOperator.AND) {
+            return cb.and(jsonPredicates);
+        } else {
+            return cb.or(jsonPredicates);
+        }
+    }
+
+    private void addUserRolePredicate(CriteriaQuery<?> query,
                                       Root<JsonSchemaDocument> documentRoot,
                                       List<Predicate> predicates) {
         List<String> roles = SecurityUtils.getCurrentUserRoles();
 
         Subquery<String> sub = query.subquery(String.class);
         Root<JsonSchemaDocumentDefinitionRole> subRoot = sub.from(JsonSchemaDocumentDefinitionRole.class);
-        sub.select(subRoot.get("id").get("documentDefinitionName"));
-        sub.where(subRoot.get("id").get("role").in(roles));
+        sub.select(subRoot.get(ID).get(DOCUMENT_DEFINITION_NAME));
+        sub.where(subRoot.get(ID).get(ROLE).in(roles));
 
         predicates.add(
-            documentRoot.get("documentDefinitionId").get("name").in(sub)
+            documentRoot.get(DOCUMENT_DEFINITION_ID).get(NAME).in(sub)
         );
     }
 
     private Predicate findJsonPathValue(CriteriaBuilder cb, Root<JsonSchemaDocument> root, String path, String value) {
-        return queryDialectHelper.getJsonValueExistsInPathExpression(cb, root.get("content").get("content"), path, value);
+        return queryDialectHelper.getJsonValueExistsInPathExpression(cb, root.get(CONTENT).get(CONTENT), path, value);
     }
 
     private Predicate findJsonValue(CriteriaBuilder cb, Root<JsonSchemaDocument> root, String value) {
-        return queryDialectHelper.getJsonValueExistsExpression(cb, root.get("content").get("content"), value);
+        return queryDialectHelper.getJsonValueExistsExpression(cb, root.get(CONTENT).get(CONTENT), value);
     }
 
-    private List<Order> getOrderBy(CriteriaBuilder cb, Root<JsonSchemaDocument> root, Sort sort) {
+    private <T extends Comparable<? super T>> Predicate buildQueryForSearchCriteria(
+        CriteriaBuilder cb,
+        Root<JsonSchemaDocument> root,
+        SearchRequest2.SearchCriteria2 searchCriteria
+    ) {
+        Expression<T> value;
+        if (searchCriteria.getPath().startsWith(DOC_PREFIX)) {
+            value = getValueExpressionForDocPrefix(cb, root, searchCriteria);
+        } else if (searchCriteria.getPath().startsWith(CASE_PREFIX)) {
+            value = getValueExpressionForCasePrefix(root, searchCriteria);
+        } else {
+            throw new IllegalArgumentException("Search path doesn't start with known prefix: '" + searchCriteria.getPath() + "'");
+        }
+
+        var rangeFrom = searchCriteria.<T>getRangeFrom();
+        var rangeTo = searchCriteria.<T>getRangeTo();
+
+        switch (searchCriteria.getSearchType()) {
+            case LIKE:
+                return cb.or(searchLike(cb, value, searchCriteria.getValues()));
```

Comment:
Why is the `cb.or` here? Do that still work with an and search?

Reply:
I guess it's to support lists of strings?

Reply:
Yes, its to support this:
```
new SearchRequest2().addOtherFilters(new SearchRequest2.SearchCriteria2()
    .path("doc:customer.fullName")
    .searchType(LIKE)
    .addValue("John D")
    .addValue("Jane D"))
```